### PR TITLE
migrate e2e auth tests to common #8

### PR DESCRIPTION
**Pull request by chaochn47**
Related to the ask from https://github.com/etcd-io/etcd/pull/14041 to split tests into smaller pieces. This PR is the last one and should close https://github.com/etcd-io/etcd/pull/14041. 

So code review will be reviewer friendly and easier to reason about.

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

/cc @ahrtr @serathius @mitake @ptabor @fuweid

### CODE REVIEW

File: tests/e2e/ctl_v3_auth_test.go [hunk 1]
```diff
@@ -303,29 +296,6 @@ func authTestWatch(cx ctlCtx) {
 
 }
 
-func authTestDefrag(cx ctlCtx) {
```
Comment (chaochn47) [inline tests/e2e/ctl_v3_auth_test.go:306]: It has been replaced by https://github.com/etcd-io/etcd/blob/main/tests/common/maintenance_auth_test.go#L33-L53

File: tests/integration/v3_auth_test.go [hunk 2]
```diff
@@ -420,83 +420,6 @@ func TestV3AuthOldRevConcurrent(t *testing.T) {
 	wg.Wait()
 }
 
-func TestV3AuthRestartMember(t *testing.T) {
```
Comment (chaochn47) [inline tests/integration/v3_auth_test.go:423]: It is the same verification as `authTestCacheReload` in `ctl_v3_auth_test.go` and it was added in https://github.com/etcd-io/etcd/pull/14358

File: tests/integration/v3_auth_test.go [hunk 3]
```diff
@@ -583,86 +506,3 @@ func TestV3AuthWatchErrorAndWatchId0(t *testing.T) {
 
 	<-watchEndCh
 }
-
-func TestV3AuthWithLeaseTimeToLive(t *testing.T) {
```
Comment (chaochn47) [inline tests/integration/v3_auth_test.go:587]: It is the same verification of `authTestLeaseTimeToLive` in `ctl_v3_auth_test.go` and was added in https://github.com/etcd-io/etcd/pull/15656

**Review by mitake — APPROVED** [review]
LGTM, thanks a lot @chaochn47

**Review by fuweid — APPROVED** [review]
LGTM

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks @chaochn47

File: tests/common/auth_test.go [hunk 4]
```diff
@@ -736,6 +736,121 @@ func TestAuthRoleList(t *testing.T) {
...
+	testutils.ExecuteUntil(ctx, t, func() {
+		require.NoErrorf(t, setupAuth(cc, []authRole{testRole}, []authUser{rootUser, testUser}), "failed to enable auth")
+		testUserAuthClient := testutils.MustClient(clus.Client(WithAuth(testUserName, testPassword)))
+
+		require.NoError(t, testUserAuthClient.Put(ctx, "foo", "bar", config.PutOptions{}))
+		// wait an expiration of my JWT token
+		<-time.After(3 * time.Second)
+
+		// e2e test will generate a new token while
+		// integration test that re-uses the same etcd client will refresh the token on server failure.
```
Comment (serathius) [inline tests/common/auth_test.go:755]: Could we validate that we refreshed token?

File: tests/common/auth_test.go [hunk 5]
```diff
@@ -736,6 +736,121 @@ func TestAuthRoleList(t *testing.T) {
...
+
+		// create foo since that is within the permission set
+		// expectation is to succeed
+		require.NoError(t, testUserAuthClient.Put(ctx, "foo", "bar", config.PutOptions{}))
+
+		// restart the node
+		clus.Members()[0].Stop()
+		require.NoError(t, clus.Members()[0].Start(ctx))
+
+		// nothing has changed, but it fails without refreshing cache after restart
```
Comment (serathius) [inline tests/common/auth_test.go:816]: Don't understand this comment. How do we know it would fail?

**Review by serathius — APPROVED** [review]
Please consider my comments for followups.

File: tests/common/auth_test.go [hunk 6]
```diff
@@ -736,6 +736,121 @@ func TestAuthRoleList(t *testing.T) {
...
+	clus := testRunner.NewCluster(ctx, t, config.WithClusterConfig(config.ClusterConfig{ClusterSize: 1, AuthToken: defaultAuthToken}))
+	defer clus.Close()
+	cc := testutils.MustClient(clus.Client())
+	testutils.ExecuteUntil(ctx, t, func() {
+		require.NoErrorf(t, setupAuth(cc, []authRole{testRole}, []authUser{rootUser, testUser}), "failed to enable auth")
+		testUserAuthClient := testutils.MustClient(clus.Client(WithAuth(testUserName, testPassword)))
+
+		require.NoError(t, testUserAuthClient.Put(ctx, "foo", "bar", config.PutOptions{}))
+		// wait an expiration of my JWT token
+		<-time.After(3 * time.Second)
```
[the remark this answers is not in the document]
Comment (chaochn47) [reply tests/common/auth_test.go:752]: It is set in `defaultAuthToken` https://github.com/etcd-io/etcd/blob/9a922091ed07c56c0c09575c2328e9576c76d942/tests/common/auth_test.go#L31

I will make the change to share the same TTL after this PR is merged.

File: tests/common/auth_test.go (hunk 4 above)
Comment (chaochn47) [reply tests/common/auth_test.go:755]: Yeah, it is feasible if 

1. etcd server log level is debug level
2. capture the etcd server log output

https://github.com/etcd-io/etcd/blob/4da39e4b1e4bc6028f69b29171e82149a72e5b37/server/auth/store.go#L339-L343

File: tests/common/auth_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (serathius) [reply tests/common/auth_test.go:752]: Sounds good

**Pull request merged**